lib/ash_json_api/json_schema/json_schema.ex

```diff
@@ -648,30 +648,32 @@ defmodule AshJsonApi.JsonSchema do
     end)
   end
 
-  defp required_write_attributes(resource) do
+  defp required_write_attributes(resource, accept) do
     resource
     |> AshJsonApi.Resource.fields()
     |> Enum.map(&Ash.Resource.attribute(resource, &1))
     |> Enum.reject(&is_nil/1)
+    |> Enum.filter(&(is_nil(accept) || &1.name in accept))
     |> Enum.filter(& &1.writable?)
     |> Enum.reject(& &1.allow_nil?)
     |> Enum.reject(& &1.default)
     |> Enum.reject(& &1.generated?)
     |> Enum.map(&to_string(&1.name))
   end
 
-  defp write_attributes(resource) do
+  defp write_attributes(resource, accept \\ []) do
```

Comment:
Why does this need to be optional?

Reply:
@zachdaniel I will be trimming everything extra out with further changes, this is probably due to fact that I started out with post changes first. I will have to check with other references of write_attributes as well.